packedForest/src/forestTypes/binnedTree/processingNodeBin.h

```diff
@@ -184,6 +208,50 @@ namespace fp{
 						}
 
 					}
+				}
+
+
+				inline void loadWorkingSet(weightedFeature& currMTRY){
+
+					typename std::vector<zipClassAndValue<int,T> >::iterator zipIterator = zipIters.returnZipBegin();
+					T accumulator;
+					int weightNum;
+
+					for(int classNum = 0; classNum < fpSingleton::getSingleton().returnNumClasses(); ++classNum){
+
+						int sizeToPrefetch = globalPrefetchSize;            
+						if(nodeIndices.returnEndIterator(classNum) - nodeIndices.returnBeginIterator(classNum) < 32){ 
+							sizeToPrefetch = nodeIndices.returnEndIterator(classNum) - nodeIndices.returnBeginIterator(classNum);
+						}
+
+						for(std::vector<int>::iterator q=nodeIndices.returnBeginIterator(classNum); q!=nodeIndices.returnBeginIterator(classNum)+sizeToPrefetch; ++q){
+							for(auto i : currMTRY.returnFeatures()){
+								fpSingleton::getSingleton().prefetchFeatureVal(i,*q);
+							}
+						}
+
+						for(std::vector<int>::iterator q=nodeIndices.returnBeginIterator(classNum); q!=nodeIndices.returnEndIterator(classNum)-sizeToPrefetch; ++q){
```

Comment:
can we just have one function with inputs of the indices to iterate over instead of these two nearly identical for loops?

Reply:
I'm not sure which loops you are referring to but I agree that if code is duplicated it can be a function... most the time.  If the work does is similar but not exact then the loop needs to remain.